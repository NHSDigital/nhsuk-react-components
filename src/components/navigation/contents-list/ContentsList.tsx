import React, { HTMLProps } from 'react';
import classNames from 'classnames';

interface ContentsListItemProps extends HTMLProps<HTMLAnchorElement> {
  current?: boolean;
}

const ContentsListItem: React.FC<ContentsListItemProps> = ({ className, current, ...rest }) => (
  <li className={classNames('nhsuk-contents-list__item', className)}>
    {current ? (
      <span className="nhsuk-contents-list__current" {...rest} />
    ) : (
      <a className="nhsuk-contents-list__link" {...rest} />
    )}
  </li>
);

interface ContentsListProps extends HTMLProps<HTMLDivElement> {
  visuallyHiddenText?: false | string;
}

interface ContentsList extends React.FC<ContentsListProps> {
  Item: React.FC<ContentsListItemProps>;
}

const ContentsList: ContentsList = ({
  className, children, visuallyHiddenText, ...rest
}) => (
  <nav className={classNames('nhsuk-contents-list', className)} {...rest}>
    {visuallyHiddenText !== false ? (
      <h2 className="nhsuk-u-visually-hidden">{visuallyHiddenText}</h2>
    ) : null}
    <ol className="nhsuk-contents-list__list">{children}</ol>
  </nav>
);

ContentsList.defaultProps = {
  role: 'navigation',
  visuallyHiddenText: 'Contents',
};

ContentsList.Item = ContentsListItem;

export default ContentsList;
