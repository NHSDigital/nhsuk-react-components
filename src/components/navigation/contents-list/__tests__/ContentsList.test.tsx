import React from 'react';
import { render } from '@testing-library/react';
import ContentsList from '../';

describe('ContentsList', () => {
  it('matches snapshot', () => {
    const { container } = render(<ContentsList />);

    expect(container).toMatchSnapshot('ContentsList');
  });

  it('renders default hidden text', () => {
    const { container } = render(<ContentsList />);

    expect(container.querySelector('.nhsuk-u-visually-hidden')?.textContent).toEqual('Contents');
  });

  it('renders custom hidden text', () => {
    const { container } = render(<ContentsList visuallyHiddenText="Custom" />);

    expect(container.querySelector('.nhsuk-u-visually-hidden')?.textContent).toEqual('Custom');
  });

  it('disables hidden text', () => {
    const { container } = render(<ContentsList visuallyHiddenText={false} />);

    expect(container.querySelector('.nhsuk-u-visually-hidden')).toBeFalsy();
  });

  describe('ContentsList.Item', () => {
    it('matches snapshot', () => {
      const { container } = render(<ContentsList.Item>Content</ContentsList.Item>);

      expect(container).toMatchSnapshot('ContentsList.Item');
    });

    it('renders as span when current', () => {
      const { container } = render(<ContentsList.Item current>Content</ContentsList.Item>);

      expect(container.querySelector('span.nhsuk-contents-list__current')?.textContent).toBe(
        'Content',
      );
    });

    it('normally renders as anchor', () => {
      const { container } = render(<ContentsList.Item>Content</ContentsList.Item>);

      expect(container.querySelector('a.nhsuk-contents-list__link')?.textContent).toBe('Content');
    });
  });
});
