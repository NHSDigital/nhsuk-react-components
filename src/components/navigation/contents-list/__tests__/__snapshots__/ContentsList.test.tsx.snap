// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentsList ContentsList.Item matches snapshot: ContentsList.Item 1`] = `
<li
  className="nhsuk-contents-list__item"
>
  <a
    className="nhsuk-contents-list__link"
  >
    Content
  </a>
</li>
`;

exports[`ContentsList matches snapshot: ContentsList 1`] = `
<nav
  className="nhsuk-contents-list"
  role="navigation"
>
  <h2
    className="nhsuk-u-visually-hidden"
  >
    Contents
  </h2>
  <ol
    className="nhsuk-contents-list__list"
  />
</nav>
`;
